packages/core/src/http-client/http-client.ts

```diff
@@ -126,6 +126,16 @@ export async function buildAxiosRequestConfig<T extends HttpRequestConfig>(
  */
 export const executeHttpRequest = execute(executeWithAxios);
 
+/**
+ * Builds a [[DestinationHttpRequestConfig]] for the given destination, merges it into the given requestConfig
+ * and executes it (using Axios). The request is also included in the object returned.
+ *
+ * @param destination - A destination or a destination name and a JWT.
+ * @param requestConfig - Any object representing an HTTP request.
+ * @returns An [[HttpRequestAndResponse]].
+ */
+export const executeHttpRequestReturnRequestAndResponse = execute(executeWithAxiosReturnRequestAndResponse);
```

Comment:
Why do want to offer this method? I guess I missed the use case - sorry for that. I understand that on the odata requests it is good for people to get the actual request and response if needed. But if people are using the `executeHttpRequest` method they have their request and response, or is the request also adjusted from axios but then I still have my reference to req if I just called `res = executeHttp(req)` I have the res and req pair.

Reply:
##### use case
This is purely for sdk internal usage.

##### background
The `executeHttpRequest` below contains two part `execute(destination, requestConfit)` and `executeWithAxios`, where
- `execute(destination, requestConfit)` builds a final request from destination and requestConfig and use the executor(currently only `axios`) to make the request.
- `executeWithAxios` uses `axios` to execute the final request
```
export const executeHttpRequest = execute(executeWithAxios)
```

The structure is nice, because it provides the flexibility when we want to support a second executor.

##### challenge
Given that the final request has to be build by us (currently in the `execute`), if we want to keep the structure, we can only touch the `executeWithAxios`. So I created a new variant.

##### alternatives
We can refactor the `executeHttpRequest`
```
export const executeHttpRequest = executeWithAxios(buildRequest(destination, requestConfit))
```
Then we have below when there is a second executor
```
executeWithSomeExecutor(buildRequest(destination, requestConfit))
```
Here is a new version for `executeHttpRequestReturnRequestAndResponse`, where we need to build the request first to avoid building twice, so the actual code is a bit more.
```
export const executeHttpRequestReturnRequestAndResponse = {
  request: buildRequest(destination, requestConfit)
  response: executeWithAxios(buildRequest(destination, requestConfit))
}
```

Reply:
ok understood.